internal/definitions/handler_tokenpool.go

```diff
@@ -41,34 +42,58 @@ func (dh *definitionHandler) handleTokenPoolBroadcast(ctx context.Context, state
 		return HandlerResult{Action: core.ActionReject}, i18n.NewError(ctx, coremsgs.MsgInvalidConnectorName, pool.Connector, "token")
 	}
 
+	org, err := dh.identity.GetMultipartyRootOrg(ctx)
+	if err != nil {
+		return HandlerResult{Action: core.ActionRetry}, err
+	}
+	isAuthor := org.DID == msg.Header.Author
+
 	pool.Message = msg.Header.ID
-	return dh.handleTokenPoolDefinition(ctx, state, pool)
+	pool.Name = pool.NetworkName
+	pool.Published = true
+	return dh.handleTokenPoolDefinition(ctx, state, pool, isAuthor)
 }
 
-func (dh *definitionHandler) handleTokenPoolDefinition(ctx context.Context, state *core.BatchState, pool *core.TokenPool) (HandlerResult, error) {
+func (dh *definitionHandler) handleTokenPoolDefinition(ctx context.Context, state *core.BatchState, pool *core.TokenPool, isAuthor bool) (HandlerResult, error) {
```

Comment:
This method has become quite complicated as it now needs to handle 3 different paths:

1. This is a local definition being created
2. This was a local definition on this node, and has now been published
3. This is a definition being created and immediately published by this node
4. This is a definition published by another node

I'm not 100% sure it handles all situations in the clearest or most optimal manner, but I think it does handle them consistently.